docs/freedesktop-quick-reference.rst

```diff
@@ -37,25 +37,53 @@ command, *Type* and *Icon*:
 
     [Desktop Entry]
     Name=Gnome Dictionary
-    Exec=org.gnome.Dictionary
+    Exec=gnome-dictionary
     Type=Application
     Icon=org.gnome.Dictionary
 
 Your desktop file should be prefixed with your application's appid and
-placed in ``/app/share/applications/``, you should use
-``desktop-file-validate`` to check your file for errors before including
-it.
+placed in ``/app/share/applications/`` within your app.
 
 Example:
 
 ::
 
     /app/share/applications/org.gnome.Dictionary.desktop
 
-You can find more information
-`here <https://wiki.archlinux.org/index.php/desktop_entries>`__. If
-interested, you can read also the full spec
-`here <https://standards.freedesktop.org/desktop-entry-spec/latest/>`__.
+It's recommended to use ``desktop-file-validate`` to check your file
+for errors before including it.
+
+A special note about the ``Exec`` line: When installing the app on a user's
+system, Flatpak will automatically transform your desktop file so that the app
+will be started through Flatpak. For example, when installing
+``org.gnome.Dictionary``, a file will be created at
+``/var/lib/flatpak/exports/share/applications/org.gnome.Dictionary.desktop``.
+In this file, the ``Exec`` line looks like this::
+
+    Exec=/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=gnome-dictionary org.gnome.Dictionary
+
+The command from the original desktop file will be part of the
+``--command`` argument to Flatpak. This means that in most cases, it
```

Comment:
Maybe worth mentioning that any additional arguments provided will be passed along. So `Exec=gnome-dictionary --lang=en` would become `Exec=/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=gnome-dictionary org.gnome.Dictionary --lang=en` so the app gets the command line args passed to it.

Reply:
`flatpak run` passes along the arguments that are not its own to the sandboxed command:
```
$ grep "Exec" ~/.local/share/flatpak/exports/share/applications/org.gnome.Notes.desktop 
Exec=/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=bijiben --file-forwarding org.gnome.Notes @@u %u @@
Exec=/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=bijiben org.gnome.Notes --new-note
```

Reply:
Hm, what happens if the command happens to accept `--branch` or `--arch` as a parameter?

Reply:
I don't know, might be worth testing it and submitting a bug report if it's an issue.

Reply:
Anything after the app-id is handled special and flatpak just passes through.

Reply:
I would assume the value of `command` is sanitized. But always worth testing.

Reply:
Ah, the app-id is the separator, that makes sense.

I updated the example in the PR with an argument.